File: packages/lynx_view_ios/ios/Classes/LynxViewRegistry.swift
import Flutter
import Foundation

/// Maps a Flutter-assigned `viewId` to the per-instance `FlutterMethodChannel`
/// used to talk to that view's Dart-side `LynxViewController`.
///
/// `FlutterBridgeModule` instances are constructed by the Lynx engine per
/// `LynxView` (not globally, despite modules being *registered* globally —
/// see the techspec's Native Module design note), and use this registry to
/// find their way back to the right Dart channel given the `LynxView`
/// they're attached to.
final class LynxViewRegistry {
    static let shared = LynxViewRegistry()

    private var channels: [Int: FlutterMethodChannel] = [:]

    private init() {}

    func register(viewId: Int, channel: FlutterMethodChannel) {
        channels[viewId] = channel
    }

    func unregister(viewId: Int) {
        channels.removeValue(forKey: viewId)
    }

    func channel(forViewId viewId: Int) -> FlutterMethodChannel? {
        channels[viewId]
    }
}

File: packages/lynx_view_android/android/src/main/kotlin/com/geektree0101/lynx_view_android/LynxViewRegistry.kt
package com.geektree0101.lynx_view_android

import io.flutter.plugin.common.MethodChannel

/**
 * Maps a Flutter-assigned `viewId` to the per-instance [MethodChannel] used
 * to talk to that view's Dart-side [LynxViewController].
 *
 * [FlutterBridgeModule] instances are constructed by the Lynx engine per
 * `LynxView`, each carrying its view's id as a constructor param, and use
 * this registry to turn that id back into the right Dart channel.
 */
internal object LynxViewRegistry {
    private val channels = mutableMapOf<Int, MethodChannel>()

    fun register(viewId: Int, channel: MethodChannel) {
        channels[viewId] = channel
    }

    fun unregister(viewId: Int) {
        channels.remove(viewId)
    }

    fun channelFor(viewId: Int): MethodChannel? = channels[viewId]
}
